storage/cache/flag.go

```diff
@@ -7,11 +7,24 @@ import (
 	flipt "github.com/markphelps/flipt/rpc"
 	"github.com/markphelps/flipt/storage"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/sirupsen/logrus"
 )
 
 const flagCachePrefix = "flag:"
 
+var (
+	flagCacheHitTotal = promauto.NewCounter(prometheus.CounterOpts{
```

Comment:
what do you think about changing the metric name to `cacheHitTotal` and then using [labels](https://prometheus.io/docs/practices/naming/#labels) to denote the 'type', like `type: flag` or something?

The reason I ask is because I plan to add evaluation caching in #100, and would like to be able to reuse this same metric for that and just change the label to `type: evaluation` or something. I think this is what Prometheus recommends in their Best Practices section.

What are your thoughts?

Reply:
@markphelps I hadn't seen #100 so thought it best to be more specific due to the lack of cache types currently provided, but in hindsight your counter label suggestions make more sense!
```
cache_hit_total (label: type=flag, cache=memory)
cache_miss_total (label: type=flag, cache=memory)
```

I'll update the `cache` counters to use the labels above and plan to tackle the histograms / gauges this weekend!